include/boost/gil/promote_integral.hpp

```diff
@@ -180,8 +177,8 @@ class promote_integral
     using type = typename detail::promote_integral::promote_to_larger
         <
             T,
-            typename boost::mpl::begin<integral_types>::type,
-            typename boost::mpl::end<integral_types>::type,
+            typename boost::mp11::mp_front<integral_types>::type,
+            typename boost::mp11::mp_at_c<integral_types, boost::mp11::mp_size<integral_types>::value -1>::type,
```

Comment:
Same here, `mp_front` and `mp_at_c` are not 1:1 for `mpl::begin` and `mpl::end`.